# 🛒 Retail Sales Analysis using Python

## Project Overview
This project analyzes retail sales data using Python and Pandas.

### Objectives
- Analyze total revenue
- Find top-performing categories and cities
- Analyze payment methods
- Identify best-selling products
- Analyze monthly revenue
- Understand order status
- Identify top customers

## 1. Import Libraries

In [ ]:
import pandas as pd


## 2. Load Dataset

In [120]:
df = pd.read_excel(r"Retail_Sales_Analysis_Dataset.xlsx")
print(df.head())    # 1st 5 Rows

  Order_ID Order_Date Correct_cust_id Customer_Name  Product_Name  \
0  ORD1001 2026-03-12         CUST001    Neha Pawar      Backpack   
1  ORD1002 2026-04-19         CUST002  Aarav Sharma  Water Bottle   
2  ORD1003 2026-02-20         CUST003    Riya Desai      Notebook   
3  ORD1004 2026-02-10         CUST004   Karan Mehta         Mouse   
4  ORD1005 2026-04-08         CUST005    Isha Gupta         Shoes   

      Category  Quantity  Unit_Price  Total_Amount        City Payment_Method  \
0  Accessories         1     1629.00       1629.00      Nashik            UPI   
1  Accessories         5      642.17       3210.85        Pune    Credit Card   
2   Stationery         5      108.64        543.20      Nagpur           Cash   
3  Electronics         1      946.59        946.59  Aurangabad    Credit Card   
4      Fashion         3     2945.41       8836.23       Thane    Net Banking   

  Order_Status  
0     Returned  
1    Completed  
2    Cancelled  
3    Completed  
4    Complete

## 3. Data Understanding

In [ ]:
    #Understand our dataset, data types
# df.shape    # How many rows and columns 
#df.columns
# df.info()   #Check data types
# df.dtypes 

In [ ]:
    #Checking Missing Values and Duplicates.
# df.isnull().sum()   #Check missing values
df.duplicated().sum()


In [ ]:
df.describe()   #statistics for our numerical columns, such as:Count,Average (mean),Minimum,Maximum,Standard deviation

## 4. Sales Analysis

In [ ]:
#We've now completed:
#💰 Total Revenue → ₹56,37,637.05
#📊 Revenue by Category
#🏙️ Revenue by City
#💳 Revenue by Payment Method
#📦 Best-selling Products
#📅 Monthly Revenue
#📋 Order Status Analysis
#💰 Average Order Value (AOV)
#   highest-value order.
#   Top Customers by Revenue

In [ ]:
    #analyze Revenue
df['Total_Amount'].sum()

np.float64(5637637.05)

In [122]:
    #Revenue by Category  display only last expression
df.groupby("Category")["Total_Amount"].sum().sort_values(ascending=False)
# df.groupby("City")["Total_Amount"].sum().sort_values(ascending=False)
# df.groupby("Payment_Method")["Total_Amount"].sum().sort_values(ascending=False)("Product_Name")["Quantity"].sum().sort_values(ascending=False)

Category
Electronics    3894988.51
Furniture      1110170.86
Fashion         445775.63
Accessories     164566.40
Stationery       22135.65
Name: Total_Amount, dtype: float64

## 5. Time & Order Analysis

In [ ]:
# Monthly Revenue
df.groupby(df["Order_Date"].dt.month_name())["Total_Amount"].sum().sort_values(ascending=False)

Order_Date
August       1116840.89
May           770744.19
February      724618.62
January       660719.85
June          660226.99
April         600750.46
July          530143.48
March         437890.64
September     135701.93
Name: Total_Amount, dtype: float64

In [ ]:
# Order Status Analysis
df["Order_Status"].value_counts()

Order_Status
Completed    188
Cancelled     58
Returned      54
Name: count, dtype: int64

In [ ]:
# Average Order Value
print("avg",df["Total_Amount"].mean())

avg 18792.123499999998


In [ ]:
# Highest Order Value
df["Total_Amount"].max()

np.float64(223955.5)

## 6. Customer Analysis

In [ ]:
#  This converts:KAVYA JOSHI → Kavya Joshi
df["Clean_Customer_Name"] = df["Customer_Name"].str.title()

In [96]:
# Top Customers by Revenue
df.groupby(["Correct_cust_id", "Clean_Customer_Name"])["Total_Amount"].sum().sort_values(ascending=False)

## 7. Key Findings

- Total revenue was ₹56,37,637.05.
- Electronics generated the highest category revenue.
- Navi Mumbai generated the highest city revenue.
- Net Banking generated the highest revenue among payment methods.
- Notebook was the best-selling product by quantity.
- August had the highest recorded monthly revenue.
- 188 orders were completed.
- Average order value was approximately ₹18,792.12.
- Isha Gupta was the highest-revenue customer.